qiskit_experiments/data_processing/nodes.py

```diff
@@ -774,6 +774,25 @@ def _reorder(self, unordered_data: np.ndarray) -> np.ndarray:
         else:
             return unordered_data.flatten()
 
+    def _reorder_iq(self, unordered_data: np.ndarray) -> np.ndarray:
+        """Reorder IQ data according to the measurement sequence."""
+
+        if unordered_data is None:
+            return unordered_data
+
+        ordered_data = []
+
+        if self._memory_allocation == ShotOrder.circuit_first:
+            for shot_idx in range(self._n_shots):
+                for circ_idx in range(self._n_circuits):
+                    ordered_data.append(unordered_data[circ_idx][shot_idx])
+        else:
+            for circ_idx in range(self._n_circuits):
+                for shot_idx in range(self._n_shots):
+                    ordered_data.append(unordered_data[circ_idx][shot_idx])
```

Comment:
Can this not be done with `.T` and optionally `flatten` like above?

Reply:
Unfortunately this does not work for the shape of the IQ data, the shape is: (`num_circuits`, `num_shots`, `num_qubits`, 2). (2 because each IQ point is a list: [I point, Q point])